### DISCUSSION TITLE
推荐性能更好的Tun2Socks模块

### DISCUSSION THREAD

**Discussion by heiher** [2dust/v2rayNG#2112] [General]
通过jni库看到作者使用的是badvpn-tun2socks，在此向作者推荐一个自己撸的tun2socks开源实现，效能比badvpn的更高。该项目的特点是在较低的CPU、内存使用下有着更高的传输性能。在RK3399这类的低性能设备上，能够跑满1Gbps物理网卡带宽，并且比跑不满的tun2socks项目有着更低的CPU使用率，那么在移动设备上应该会更加节能，续航友好。

项目地址: https://github.com/heiher/hev-socks5-tunnel
Android VPN参考实现: https://github.com/heiher/sockstun

#### Speed

![rk3399-speed](https://user-images.githubusercontent.com/1407733/221888472-9c4c790f-b193-4926-814c-a87a06b703bf.png)

#### CPU Usage

![rk3399-cpu](https://user-images.githubusercontent.com/1407733/221888507-512255a2-ed3c-457a-9762-821ee0ba93bd.png)

### 7950X

See [here](https://github.com/heiher/hev-socks5-tunnel/wiki/Benchmarks) for more details.

#### Speed

![](https://github.com/heiher/hev-socks5-tunnel/wiki/res/upload-speed.png)
![](https://github.com/heiher/hev-socks5-tunnel/wiki/res/download-speed.png)

#### CPU usage

![](https://github.com/heiher/hev-socks5-tunnel/wiki/res/upload-cpu.png)
![](https://github.com/heiher/hev-socks5-tunnel/wiki/res/download-cpu.png)

#### Memory usage

![](https://github.com/heiher/hev-socks5-tunnel/wiki/res/upload-mem.png)
![](https://github.com/heiher/hev-socks5-tunnel/wiki/res/download-mem.png)

**Comment by 2dust** [comment]
感谢
如更换模块，需要花时间做适配开发和稳定性测试
欢迎pr
